### PR TITLE
Catching gRPC error in datastore run_query and converting to our error.

### LINKED ISSUE #2494: Datastore fetch needs to encapsulate all gRPC errors
This code:

```python
    try:
        start_date = datetime.datetime(1990, 1, 1)
        query = client.query(kind='Task')
        query.add_filter(
            'created', '>', start_date)
        query.add_filter(
            'priority', '>', 3)

        return list(query.fetch())

    except google.cloud.exceptions.BadRequest:
        pass
```

Raises a `grpc._channel._Rendezvous` exception instead of an expected subclass of `google.cloud.exceptions.BadRequest` (I'd also accept `google.cloud.exceptions.GoogleCloudError`):

```
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.INVALID_ARGUMENT, Cannot have inequality filters on multiple properties: [created, priority])>
```

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #2494.

### CODE REVIEW

**Comment by dhermes** [comment]
It may be worth putting a try/except around every method and using the mapping provided in [`code.proto`][1]

[1]: https://github.com/googleapis/googleapis/blob/034b4cdaacf13764f70ba55205ea6f9734652cc0/google/rpc/code.proto

**Comment by jonparrott** [comment]
> It may be worth putting a try/except around every method and using the mapping provided in code.proto

Yeah, this LGTM to me as a temporary fix, but I feel it'd be more thorough to do that.

**Comment by dhermes** [comment]
Filed #2497

File: datastore/unit_tests/test_connection.py [hunk 1]
```diff
@@ -200,6 +200,43 @@ def test_run_query(self):
...
+
+    @unittest.skipUnless(_HAVE_GRPC, 'No gRPC')
+    def test_run_query_cancelled(self):
+        from grpc import StatusCode
+        from grpc._channel import _RPCState
+        from google.cloud.exceptions import GrpcRendezvous
+
+        exc_state = _RPCState((), None, None, StatusCode.CANCELLED, None)
+        exc = GrpcRendezvous(exc_state, None, None, None)
+        self._run_query_failure_helper(exc, GrpcRendezvous)
```
Comment (tseaver) [inline datastore/unit_tests/test_connection.py]: Should we be mapping a different error class for `StatusCode.CANCELLED`?)

**Comment by dhermes** [comment]
@tseaver Sorry merged based on @jonparrott LGTM. Happy to revert / quick fix if you have concerns

**Comment by tseaver** [comment]
No worries.

**Pull request merged**